node-packages/commons/src/notificationCommons.ts

```diff
@@ -0,0 +1,41 @@
+import * as R from 'ramda';
+
+// Note: we're using integer representations of the
+// severity in order to be able to use numeric comparisons
+// when determining which messages to send
+
+export const notificationContentTypeToInt = R.cond([
+    [R.equals('NONE'), R.always(0)],
+    [R.equals('UNKNOWN'), R.always(10)],
+    [R.equals('NEGLIGIBLE'), R.always(20)],
+    [R.equals('LOW'), R.always(30)],
+    [R.equals('MEDIUM'), R.always(40)],
+    [R.equals('HIGH'), R.always(50)],
+    [R.equals('CRITICAL'), R.always(60)],
+  ]);
+
+export const notificationIntToContentType = R.cond([
+    [R.equals('0'), R.always('NONE')],
+    [R.equals('10'), R.always('UNKNOWN')],
+    [R.equals('20'), R.always('NEGLIGIBLE')],
+    [R.equals('30'), R.always('LOW')],
+    [R.equals('40'), R.always('MEDIUM')],
+    [R.equals('50'), R.always('HIGH')],
+    [R.equals('60'), R.always('CRITICAL')],
+    [R.T, R.always('NONE')],
+  ]);
+
+export const parseProblemNotification = (event) => {
+  const rawArray = event.split(":");
+  const eventData = {
+    isProblem: rawArray[0] == "problem",
+    eventType: undefined,
+    dataType: undefined,
+    source: undefined,
+    severityLevel: undefined,
+  }
+  if(eventData.isProblem) {
+    [, eventData.eventType, eventData.source, eventData.dataType, eventData.severityLevel] = rawArray;
```

Comment:
I'm not sure how the event string is defined (but assume its something like `'problem:insert:drutiny:data:CRITICAL'` etc) that gets split into that rawArray. I've not seen a comma at the start of a de-structured array before, but is that meant to be 'eventData.isProblem' or is it ignoring/skipping the first value purpose? I would say add in the isProblem property also anyways perhaps.

For example:

`[eventData.isProblem, eventData.eventType, eventData.source, eventData.dataType, eventData.severityLevel] = rawArray;`

Reply:
Thanks Tim. **isProblem** is a boolean, but you'll see that I've updated it to include **eventClass**, which'll make it less obtuse